File: 2707-merge-two-2d-arrays-by-summing-values/merge-two-2d-arrays-by-summing-values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        # [[1,2], [2,3], [4,5]]
        # [[1,2], [3,3], [4,5], [5,5], [6,6]]
        result_arr = []
        obj = {}

        for i in range(len(nums1)):
            narr = nums1[i]
            obj[narr[0]] = narr[1]

        print(obj)
        for i in range(len(nums2)):
            narr = nums2[i]
            obj[narr[0]] = (obj.get(narr[0]) or 0) + narr[1]

        print(obj)
        for i,val in sorted(obj.items()):
            result_arr.append([i,val])

        return result_arr

File: 2707-merge-two-2d-arrays-by-summing-values/merge-two-2d-arrays-by-summing-values.js
/**
 * @param {number[][]} nums1
 * @param {number[][]} nums2
 * @return {number[][]}
 */
var mergeArrays = function(nums1, nums2) {
    // [[1,2], [2,3], [4,5]]
    // [[1,2], [3,3], [4,5], [5,5], [6,6]]
    let len_nums1 = nums1.length
    let len_nums2 = nums2.length
    let result_arr = []

    const obj = {}
    for (let i = 0; i < nums1.length; i++) {
        let narr = nums1[i]
        obj[narr[0]] = narr[1]
    }

    for (let i = 0; i < nums2.length; i++) {
        let narr = nums2[i]
        obj[narr[0]] = (obj[narr[0]] || 0) + narr[1]
    }
    
    for (const key in obj) {
        result_arr.push([parseInt(key), obj[key]])
    }

    return result_arr
};